pkg/cgroup/resolve_container.go

```diff
@@ -40,29 +41,196 @@ type ContainerInfo struct {
 
 const (
 	unknownPath string = "unknown"
-
-	procPath   string = "/proc/%d/cgroup"
-	cgroupPath string = "/sys/fs/cgroup"
+	procPath    string = "/proc/%d/cgroup"
+	cgroupPath  string = "/sys/fs/cgroup"
 )
 
 var (
-	byteOrder binary.ByteOrder         = utils.DetermineHostByteOrder()
-	podLister kubelet.KubeletPodLister = kubelet.KubeletPodLister{}
+	instance *cache
+	once     sync.Once
+)
 
+type cache struct {
 	// map to cache data to speedup lookups
-	containerIDCache           = map[uint64]string{}
-	containerIDToContainerInfo = map[string]*ContainerInfo{}
-	cGroupIDToPath             = map[uint64]string{}
-
-	validContainerIDRegex         = regexp.MustCompile("^[a-zA-Z0-9]+$")
-	regexReplaceContainerIDPrefix = regexp.MustCompile(`.*//`)
-)
+	containerIDCache           map[uint64]string
+	containerIDToContainerInfo map[string]*ContainerInfo
+	cGroupIDToPath             map[uint64]string
+	byteOrder                  binary.ByteOrder
+	mu                         sync.RWMutex
+}
 
 func Init() (*[]corev1.Pod, error) {
 	pods := []corev1.Pod{}
 	return &pods, nil
 }
 
+// InitCache creates the singleton Config instance if necessary.
+func InitCache() {
+	once.Do(func() {
+		instance = newCache()
+	})
+}
+
+func GetCache() *cache {
+	return instance
+}
+
+// newConfig creates and returns a new Config instance.
+func newCache() *cache {
+	return &cache{
+		containerIDCache:           map[uint64]string{},
+		containerIDToContainerInfo: map[string]*ContainerInfo{},
+		cGroupIDToPath:             map[uint64]string{},
+		byteOrder:                  utils.DetermineHostByteOrder(),
+	}
+}
```

Comment:
Given that the usage of the package is to use the `Instance` which in the current implementation should be initialised by calling `InitCache`, can we make use of `func init` to initialise the `Instance` without requiring the caller to `InitCache` ? 

```
func init() {
  instance = newCache()
}
```